src/atlas/mesh/actions/GetCubedSphereNodalArea.cc

```diff
@@ -0,0 +1,56 @@
+/*
+ * (C) Crown Copyright 2024 Met Office
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+
+#include "atlas/array/MakeView.h"
+#include "atlas/grid/CubedSphereGrid.h"
+#include "atlas/mesh/actions/GetCubedSphereNodalArea.h"
+#include "atlas/util/NormaliseLongitude.h"
+#include "atlas/util/Point.h"
+
+
+namespace atlas {
+namespace mesh {
+namespace actions {
+
+
+Field& GetCubedSphereNodalArea::operator()(Mesh& mesh) {
+
```

Comment:
What's missing here is a check  if "grid_cell_areas" field exists in the mesh.nodes(), and return it if it exists, otherwise continue computing and adding it to the mesh.nodes() as you did already.